src/output.rs

```diff
@@ -316,10 +316,35 @@ pub(crate) fn write_buffer(buffer: &[u8], f: &mut fmt::Formatter<'_>) -> fmt::Re
     if let Ok(buffer) = str::from_utf8(buffer) {
         write!(f, "{}", buffer)
     } else {
-        write!(f, "{:?}", buffer)
+        write_bytes(buffer, f)
     }
 }
 
+pub(crate) fn write_bytes(buffer: &[u8], f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    write!(f, "b\"")?;
+    for &b in buffer {
+        let repl = match b {
+            b'\n' => "\\n",
+            b'\r' => "\\r",
+            b'\t' => "\\t",
+            b'\\' => "\\\\",
+            b'\"' => "\\\"",
+            b'\0' => "\\0",
+            0x20..=0x7e => {
+                // ASCII printable characters
+                write!(f, "{}", b as char)?;
```

Comment:
Should we instead use `from_utf8_lossy`?

Reply:
I don't think so. Sometimes binary data is binary data. Here is how this commit prints first 30 bytes of mp4 file:
```
b"\0\0\0 ftypisom\0\0\x02\0isomiso2avc1mp"
```
And here is how utf8_lossy does the same:
```
 ftypisomisomiso2avc1mp
```

Reply:
Good point.  Digging into this further, we have some options for thos.

- Use `Bstr`'s `Debug`, see https://docs.rs/bstr/0.2.13/src/bstr/impls.rs.html#329-342
- Implement it ourselves using `escape_debug`, see https://doc.rust-lang.org/std/primitive.char.html#method.escape_debug
- Keep the current implementation

I'm favoring reusing BStr's `Debug`.  Does it work for your needs?

Reply:
`escape_debug` works only for characters. It could lead to misleading results if we use cast `byte as char`.

`bstr` would work but:
1. Adding a dependency instead of 5 lines of code doesn't sound good
2. It yields `"\u{0}\u{0}\u{0} ftypisom\u{0}\u{0}\u{2}\u{0}isomiso2avc1mp"` for the line above. I.e. better than status quo, but longer and less compatible than what is initially in this PR (e.g. `\u{..}` is only supported in Rust as far as I know, but `\0`, `\x02` works in Rust, C, Python, Bash, to name a few).

Reply:
Also, let's see if I can convince bstr authors: https://github.com/BurntSushi/bstr/issues/56

Reply:
> Adding a dependency instead of 5 lines of code doesn't sound good

For me, the issue is with escaping and providing accurate-enough representations.  That is an area that feels like its easy to get wrong and would rather rely on another crate.

Sounds like BurntSushi is aligned.  We can merge as-is (well, once other issues are resolved) or switch to BStr now and just upgrade to get the improved repr once it comes out.

Reply:
Bstr accepted the change, so just relying on that.